Add autopkgtests for compiler and onedpl

frenchwr opened the pull request:
This adds autopkgtests for the compiler and oneDPL library. The tests themselves are all taken from upstream and are the same ones we have used for validation until now.

I have run the tests on `mustang` using:

```
autopkgtest -B -- null
```

## Compiler test results

```
$ autopkgtest -B -- null
autopkgtest [02:21:06]: starting date and time: 2025-10-10 02:21:06+0000
autopkgtest [02:21:06]: version 5.47
autopkgtest [02:21:06]: host mustang-118102; command line: /usr/bin/autopkgtest -B -- null
autopkgtest [02:21:06]: testbed dpkg architecture: amd64
autopkgtest [02:21:06]: testbed apt version: 3.0.0
autopkgtest [02:21:06]: @@@@@@@@@@@@@@@@@@@@ test bed setup
autopkgtest [02:21:06]: testbed release detected to be: None
autopkgtest [02:21:06]: testbed running kernel: Linux 6.14.0-29-generic #29-Ubuntu SMP PREEMPT_DYNAMIC Thu Aug  7 18:32:38 UTC 2025
autopkgtest [02:21:06]: @@@@@@@@@@@@@@@@@@@@ unbuilt-tree .
autopkgtest [02:21:06]: testing package intel-dpcpp version 6.2.0-0ubuntu1~25.10~ppa7
autopkgtest [02:21:06]: build not needed
autopkgtest [02:21:06]: test sycl-neo-device: preparing testbed
autopkgtest [02:21:07]: test sycl-neo-device: [-----------------------
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/sycl-neo-device] Starting build phase
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/sycl-neo-device] Starting run phase
No device of requested type available.
autopkgtest [02:21:10]: test sycl-neo-device: -----------------------]
autopkgtest [02:21:10]: test sycl-neo-device:  - - - - - - - - - - results - - - - - - - - - -
sycl-neo-device      PASS
autopkgtest [02:21:10]: test mandelbrot: preparing testbed
autopkgtest [02:21:10]: test mandelbrot: [-----------------------
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/mandelbrot] Starting build phase
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Default CMAKE_BUILD_TYPE not set using Release with Debug Info
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is Clang 21.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/002-mandelbrot
[ 25%] Building CXX object src/CMakeFiles/mandelbrot.dir/main.cpp.o
[ 50%] Linking CXX executable mandelbrot
[ 50%] Built target mandelbrot
[ 75%] Building CXX object src/CMakeFiles/mandelbrot_usm.dir/main.cpp.o
[100%] Linking CXX executable mandelbrot_usm
[100%] Built target mandelbrot_usm
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/mandelbrot] Standard run
     Platform Name: Intel(R) oneAPI Unified Runtime over Level-Zero
  Platform Version: 1.6
       Device Name: Intel(R) Arc(TM) B580 Graphics
    Max Work Group: 1024
 Max Compute Units: 160

Parallel Mandelbrot set using buffers.
 Rendered image output to file: mandelbrot.png (output too large to display in text)
       Serial time: 0.0265702s
     Parallel time: 0.00994707s
Successfully computed Mandelbrot set.
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/mandelbrot] Unified shared memory (USM) run
     Platform Name: Intel(R) oneAPI Unified Runtime over Level-Zero
  Platform Version: 1.6
       Device Name: Intel(R) Arc(TM) B580 Graphics
    Max Work Group: 1024
 Max Compute Units: 160

Parallel Mandelbrot set using USM.
 Rendered image output to file: mandelbrot.png (output too large to display in text)
       Serial time: 0.026055s
     Parallel time: 4.07476e-05s
Successfully computed Mandelbrot set.
autopkgtest [02:21:23]: test mandelbrot: -----------------------]
autopkgtest [02:21:23]: test mandelbrot:  - - - - - - - - - - results - - - - - - - - - -
mandelbrot           PASS
autopkgtest [02:21:23]: test vector-add: preparing testbed
autopkgtest [02:21:23]: test vector-add: [-----------------------
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/vector-add] Starting build phase - buffer implementation
CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 21.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/003-vector-add/buffers
[ 50%] Building CXX object src/CMakeFiles/vector-add-buffers.dir/vector-add-buffers.cpp.o
[100%] Linking CXX executable ../vector-add-buffers
[100%] Built target vector-add-buffers
[100%] Built target cpu-gpu
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/vector-add] Starting run phase - buffer implementation
Running on device: Intel(R) Arc(TM) B580 Graphics
Vector size: 10000
[0]: 0 + 0 = 0
[1]: 1 + 1 = 2
[2]: 2 + 2 = 4
...
[9999]: 9999 + 9999 = 19998
Vector add successfully completed on device.
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/vector-add] Starting build phase - USM implementation
CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 21.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-dpcpp - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using the USM variant.
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/003-vector-add/usm
[ 50%] Building CXX object src/CMakeFiles/vector-add-usm.dir/vector-add-usm.cpp.o
[100%] Linking CXX executable ../vector-add-usm
[100%] Built target vector-add-usm
[100%] Built target cpu-gpu
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/vector-add] Starting run phase - USM implementation
Running on device: Intel(R) Arc(TM) B580 Graphics
Vector size: 10000
[0]: 0 + 0 = 0
[1]: 1 + 1 = 2
[2]: 2 + 2 = 4
...
[9999]: 9999 + 9999 = 19998
Vector add successfully completed on device.
autopkgtest [02:21:31]: test vector-add: -----------------------]
autopkgtest [02:21:31]: test vector-add:  - - - - - - - - - - results - - - - - - - - - -
vector-add           PASS
autopkgtest [02:21:31]: test matrix-mul: preparing testbed
autopkgtest [02:21:31]: test matrix-mul: [-----------------------
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/matrix-mul] Starting build phase
clang++-dpcpp  -std=c++17 -fsycl -g -o matrix_mul_dpc src/matrix_mul_sycl.cpp 
>>> [/tmp/autopkgtest.zhugia/build.iR1/real-tree/debian/tests/matrix-mul] Starting run phase
Device: Intel(R) Arc(TM) B580 Graphics
Problem size: c(150,600) = a(150,300) * b(300,600)
Result of matrix multiplication using SYCL: Success - The results are correct!
autopkgtest [02:21:35]: test matrix-mul: -----------------------]
autopkgtest [02:21:35]: test matrix-mul:  - - - - - - - - - - results - - - - - - - - - -
matrix-mul           PASS
autopkgtest [02:21:35]: @@@@@@@@@@@@@@@@@@@@ summary
sycl-neo-device      PASS
mandelbrot           PASS
vector-add           PASS
matrix-mul           PASS
```

## oneDPL test results

```
$ autopkgtest -B -- null
autopkgtest [02:45:22]: starting date and time: 2025-10-10 02:45:22+0000
autopkgtest [02:45:22]: version 5.47
autopkgtest [02:45:22]: host mustang-118102; command line: /usr/bin/autopkgtest -B -- null
autopkgtest [02:45:22]: testbed dpkg architecture: amd64
autopkgtest [02:45:22]: testbed apt version: 3.0.0
autopkgtest [02:45:22]: @@@@@@@@@@@@@@@@@@@@ test bed setup
autopkgtest [02:45:22]: testbed release detected to be: None
autopkgtest [02:45:22]: testbed running kernel: Linux 6.14.0-29-generic #29-Ubuntu SMP PREEMPT_DYNAMIC Thu Aug  7 18:32:38 UTC 2025
autopkgtest [02:45:22]: @@@@@@@@@@@@@@@@@@@@ unbuilt-tree .
autopkgtest [02:45:22]: testing package onedpl version 2022.9.0-0ubuntu1~25.10~ppa6
autopkgtest [02:45:22]: build not needed
autopkgtest [02:45:22]: test reduce-segment: preparing testbed
autopkgtest [02:45:22]: test reduce-segment: [-----------------------
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/reduce-segment] Starting build phase
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/reduce-segment] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
Keys = [ 0 0 0 1 1 1 ]
Values = [ 1 2 3 4 5 6 ]
Output Keys = [ 0 1 ]
Output Values = [ 6 15 ]
autopkgtest [02:45:28]: test reduce-segment: -----------------------]
autopkgtest [02:45:28]: test reduce-segment:  - - - - - - - - - - results - - - - - - - - - -
reduce-segment       PASS
autopkgtest [02:45:28]: test binary-search: preparing testbed
autopkgtest [02:45:28]: test binary-search: [-----------------------
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/binary-search] Starting build phase
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/binary-search] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
Input sequence = [0 2 2 2 3 3 3 3 6 6 ]
Search sequence = [0 2 4 7 6 ]
Search results = [1 1 0 0 1 ]
autopkgtest [02:45:33]: test binary-search: -----------------------]
autopkgtest [02:45:33]: test binary-search:  - - - - - - - - - - results - - - - - - - - - -
binary-search        PASS
autopkgtest [02:45:33]: test dpl-sortdouble: preparing testbed
autopkgtest [02:45:33]: test dpl-sortdouble: [-----------------------
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/dpl-sortdouble] Starting build phase
>>> [/tmp/autopkgtest.kiOIET/build.Qio/real-tree/debian/tests/dpl-sortdouble] Starting run phase
Device : Intel(R) Arc(TM) B580 Graphics
2
4
6
8
autopkgtest [02:45:41]: test dpl-sortdouble: -----------------------]
autopkgtest [02:45:42]: test dpl-sortdouble:  - - - - - - - - - - results - - - - - - - - - -
dpl-sortdouble       PASS
autopkgtest [02:45:42]: @@@@@@@@@@@@@@@@@@@@ summary
reduce-segment       PASS
binary-search        PASS
dpl-sortdouble       PASS
```

frenchwr committed:
Add autopkgtests for compiler and onedpl

hector-cao commented on compiler/debian/tests/control:
```diff
@@ -0,0 +1,5 @@
+Tests: sycl-neo-device, mandelbrot, vector-add, matrix-mul
+# libstb-dev required by mandelbrot test
+# ocl-icd-opencl-dev required by mandebrot test
+Depends: @, libstb-dev, ocl-icd-opencl-dev
```
in some tests, we need cmake, probably, you already have it in the testbed but i think it is better to specify this dependency explicitly

frenchwr replied:
Good call. I just added `cmake` and also realized I was missing the compiler dependency for the oneDPL tests.

frenchwr committed:
Add missing test dependencies

hector-cao reviewed (dismissed):
LGTM :rocket:

frenchwr committed:
Remove unified-runtime package.

Starting with the 6.2.0 compiler release, unified runtime
is now included directly in the source tree for the compiler.

frenchwr committed:
Remove compiler samples

frenchwr committed:
compiler: add autopkgtests to source package

frenchwr committed:
onedpl: add autopkgtests to source package

frenchwr committed:
Update gitignore entries

frenchwr commented:
@hector-cao I made a few updates if you wouldn't mind reviewing one last time:

* Rebase against tip of main
* Release new compiler and onedpl packages in PPA with autopkgtests
* Remove unified-runtime package since that is now included in the compiler source tree
* Remove various validation and samples that we used in the early days of this repo, but these are no longer needed now that we have added autopkgtests

Those last two made the PR quite large, but this will make the repo much cleaner and easier for us to maintain.
